autotests/testolmaccount.cpp

```diff
@@ -0,0 +1,583 @@
+              "\"device_id\":\"JLAFKJWSCS\",\"keys\":{\"curve25519:JLAFKJWSCS\":"
+              "\"3C5BFWi2Y8MaVvjM8M22DBmh24PmgR0nPvJOIArzgyI\",\"ed25519:JLAFKJWSCS\":"
+              "\"lEuiRJBit0IG6nUf5pUzWTUEsRVVe/"
+              "HJkoKuEww9ULI\"},\"signatures\":{\"@alice:example.com\":{\"ed25519:JLAFKJWSCS\":"
+              "\"dSO80A01XiigH3uBiDVx/EjzaoycHcjq9lfQX0uWsqxl2giMIiSPR8a4d291W1ihKJL/"
+              "a+myXS367WT6NAIcBA\"}},\"user_id\":\"@alice:example.com\"}");
+
+    auto doc2 = QJsonDocument::fromJson(R"({
+      "user_id": "@alice:example.com",
+      "device_id": "JLAFKJWSCS",
+      "algorithms": [
+        "m.olm.v1.curve25519-aes-sha2",
+        "m.megolm.v1.aes-sha2"
+      ],
+      "keys": {
+        "curve25519:JLAFKJWSCS": "3C5BFWi2Y8MaVvjM8M22DBmh24PmgR0nPvJOIArzgyI",
+        "ed25519:JLAFKJWSCS": "lEuiRJBit0IG6nUf5pUzWTUEsRVVe/HJkoKuEww9ULI"
+      },
+      "signatures": {
+        "@alice:example.com": {
+          "ed25519:JLAFKJWSCS": "dSO80A01XiigH3uBiDVx/EjzaoycHcjq9lfQX0uWsqxl2giMIiSPR8a4d291W1ihKJL/a+myXS367WT6NAIcBA"
+        }
+      },
+      "unsigned": {
+        "device_display_name": "Alice's mobile phone"
+      }
+    })");
+
+    DeviceKeys device2;
+    JsonObjectConverter<DeviceKeys>::fillFrom(doc2.object(), device2);
+
+    QCOMPARE(device2.userId, device1.userId);
+    QCOMPARE(device2.deviceId, device1.deviceId);
+    QCOMPARE(device2.keys, device1.keys);
+    QCOMPARE(device2.algorithms, device1.algorithms);
+    QCOMPARE(device2.signatures, device1.signatures);
+
+    // UnsignedDeviceInfo is missing from the generated DeviceKeys object :(
+    // QCOMPARE(device2.unsignedInfo.deviceDisplayName, "Alice's mobile phone");
+}
+
+void TestOlmAccount::encryptedFile()
+{
+    auto doc = QJsonDocument::fromJson(R"({
+      "url": "mxc://example.org/FHyPlCeYUSFFxlgbQYZmoEoe",
+      "v": "v2",
+      "key": {
+        "alg": "A256CTR",
+        "ext": true,
+        "k": "aWF6-32KGYaC3A_FEUCk1Bt0JA37zP0wrStgmdCaW-0",
+        "key_ops": ["encrypt","decrypt"],
+        "kty": "oct"
+      },
+      "iv": "w+sE15fzSc0AAAAAAAAAAA",
+      "hashes": {
+        "sha256": "fdSLu/YkRx3Wyh3KQabP3rd6+SFiKg5lsJZQHtkSAYA"
+      }})");
+
+    EncryptedFile file;
+    JsonObjectConverter<EncryptedFile>::fillFrom(doc.object(), file);
+
+    QCOMPARE(file.v, "v2");
+    QCOMPARE(file.iv, "w+sE15fzSc0AAAAAAAAAAA");
+    QCOMPARE(file.hashes["sha256"], "fdSLu/YkRx3Wyh3KQabP3rd6+SFiKg5lsJZQHtkSAYA");
+    QCOMPARE(file.key.alg, "A256CTR");
+    QCOMPARE(file.key.ext, true);
+    QCOMPARE(file.key.k, "aWF6-32KGYaC3A_FEUCk1Bt0JA37zP0wrStgmdCaW-0");
+    QCOMPARE(file.key.keyOps.count(), 2);
+    QCOMPARE(file.key.kty, "oct");
+}
+
+#define CREATE_CONNECTION(VAR, USERNAME, SECRET, DEVICE_NAME) \
+    NetworkAccessManager::instance()->ignoreSslErrors(true); \
+    auto VAR = std::make_shared<Connection>(); \
+    (VAR) ->resolveServer("@alice:localhost:" + QString::number(443)); \
```

Comment:
There's a null character right after `+`; but otherwise, why complicate things like this even?
```suggestion
    (VAR) ->resolveServer("@alice:localhost:443"); \
```